models/DataObject/ClassDefinition.php

```diff
@@ -397,12 +405,30 @@ public function save($saveDefinitionFile = true)
         }
         $cd .= "*/\n\n";
 
-        $implementsBlock = '\\Pimcore\\Model\\DataObject\\DirtyIndicatorInterface';
+        $implementsParts = ['\\Pimcore\\Model\\DataObject\\DirtyIndicatorInterface'];
+
         if ($this->getCacheRawRelationData()) {
-            $implementsBlock .= ',\\Pimcore\\Model\\DataObject\\CacheRawRelationDataInterface';
+            $implementsParts[] = '\\Pimcore\\Model\\DataObject\\CacheRawRelationDataInterface';
+        }
+
+        if ($this->getImplementsInterfaces()) {
+            $customParts = $this->getImplementsInterfaces();
+            $customParts = explode(',', $customParts);
+            foreach ($customParts as $interface) {
+                $interface = trim($interface);
+                if (Tool::interfaceExists($interface)) {
+                    $customParts[]= $interface;
+                } else {
+                    throw new \Exception("interface '" . $interface . "' does not exist");
+                }
+            }
+
+            $implementsParts[] = $this->getImplementsInterfaces();
```

Comment:
This is somehow strange.

Is the `foreach` only for checking the existence? In that case line 420 `$customParts[]= $interface;` is not needed.

Otherwise, the line 426 `$implementsParts[] = $this->getImplementsInterfaces();` should be removed and line 420 should be `$implementsParts[]= $interface;` instead.